esphome/components/wifi_info/wifi_info_text_sensor.h

```diff
@@ -24,6 +24,35 @@ class IPAddressWiFiInfo : public Component, public text_sensor::TextSensor {
   IPAddress last_ip_;
 };
 
+class ScanResultsWiFiInfo : public Component, public text_sensor::TextSensor {
+ public:
+  void loop() override {
```

Comment:
```suggestion
  void update() override {
```

This calculation shouldn't happen in `loop()` - it's constructing a long string with many memory allocations. With `loop` that will run 60 times per second.

Reply:
Done. Thanks.

Reply:
Actually, I just tried to update one of my devices and I'm getting an error compiling:
```
In file included from src/esphome/components/wifi_info/wifi_info_text_sensor.cpp:1:0:
src/esphome/components/wifi_info/wifi_info_text_sensor.h:29:8: error: 'void esphome::wifi_info::ScanResultsWiFiInfo::update()' marked 'override', but does not override
   void update() override {
        ^
```
@OttoWinter I believe I didn't understand your suggestion fully. I thought that `update` was a method I could override and only ran in certain situations but guess not.

Reply:
You should change the `ScanResultsWiFiInfo` class to inherit from `PollingComponent`.

Reply:
@oxan Thanks, worked perfectly.
@OttoWinter suggestion applied.